ios/Exponent/Versioned/Modules/Api/EXContacts.m

```diff
@@ -54,13 +67,54 @@ @implementation EXContacts
 
     NSMutableDictionary *contact = [NSMutableDictionary dictionary];
     contact[@"id"] = person.identifier;
+    contact[@"contactType"] = person.contactType == CNContactTypePerson ? @"person" : @"organization";
+    contact[@"name"] = [self _assembleDisplayNameForContact:person];
     contact[@"firstName"] = person.givenName;
     contact[@"lastName"] = person.familyName;
+    contact[@"previousLastName"] = person.previousFamilyName;
     contact[@"middleName"] = person.middleName;
-    contact[@"name"] = [self _assembleDisplayNameForContact:person];
+    contact[@"nickname"] = person.nickname;
     contact[@"company"] = person.organizationName;
     contact[@"jobTitle"] = person.jobTitle;
+    contact[@"department"] = person.departmentName;
+    contact[@"imageAvailable"] = [NSNumber numberWithBool:person.imageDataAvailable];
 
+    if ([keysToFetch containsObject:CNContactNamePrefixKey]) {
+      contact[@"namePrefix"] = person.namePrefix;
+    }
+    if ([keysToFetch containsObject:CNContactNameSuffixKey]) {
+      contact[@"nameSuffix"] = person.nameSuffix;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticGivenNameKey]) {
+      contact[@"phoneticFirstName"] = person.phoneticGivenName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticMiddleNameKey]) {
+      contact[@"phoneticMiddleName"] = person.phoneticMiddleName;
+    }
+    if ([keysToFetch containsObject:CNContactPhoneticFamilyNameKey]) {
+      contact[@"phoneticLastName"] = person.phoneticFamilyName;
+    }
+    if ([keysToFetch containsObject:CNContactNoteKey]) {
+      contact[@"note"] = person.note;
+    }
+    if ([keysToFetch containsObject:CNContactImageDataKey]) {
+      contact[@"image"] = @{ @"uri": [NSString stringWithFormat:@"%@%@",
```

Comment:
I'd rather keep the API the same for both android and iOS. We should perhaps add an appropriate comment in docs so that ppl know who can happen if they request the image field to be returned in the contacts data. Note that we don't send the image data if it hasn't been requested.

I don't see any way on how this could be implemented on iOS without passing the image data over the bridge. The issue is that the data is returned from the iOS API as a part of the contacts record in NSData. There is no way currently for us to retain that data on the native side as even if we do we won't be able to tell when the data is no longer being referenced on the JS side and when it should be released.

One idea I have at the moment would be to introduce a native component "ContactImage" that would take ID of the contact as a parameter. It could extend the regular "Image" component but instead of loading image data from a given source it would use contacts API to get the image data and to get it displayed